### PR TITLE
Implement pylint's `too-many-statements` rule (`PLR0915`)

### PR DESCRIPTION
**Pull request by chanman3388**
https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/too-many-statements.html

A bit of a quirky one, I made a bunch of test cases in python to see how pylint would work (tested on 2.9.3) to derive the expected behaviour.

Issue #970

### CODE REVIEW

**Comment by chanman3388** [comment]
I'm unsure how to fix the unittest as it passes for me locally:
```
➜  ruff git:(PLR0915) cargo test rules::pylint::tests::max_statements                    
    Finished test [unoptimized + debuginfo] target(s) in 0.84s
     Running unittests src/lib.rs (target/debug/deps/ruff-6394b47557a9e4c6)

running 1 test
test rules::pylint::tests::max_statements ... ok
```

File: src/rules/pylint/settings.rs [hunk 1]
```diff
@@ -56,11 +56,7 @@ pub struct Options {
     #[option(default = r"5", value_type = "int", example = r"max-args = 5")]
     /// Maximum number of arguments allowed for a function definition (see: `PLR0913`).
     pub max_args: Option<usize>,
-    #[option(
-        default = r"100",
-        value_type = "int",
-        example = r"max_statements = 100"
-    )]
+    #[option(default = r"50", value_type = "int", example = r"max_statements = 50")]
```
Comment (charliermarsh) [inline src/rules/pylint/settings.rs:59]: Nit: in the example, use `max-statements`, since we use that casing in the TOML.

Comment (chanman3388) [reply src/rules/pylint/settings.rs:59]: Thank you

**Comment by charliermarsh** [comment]
Can't for the life of me figure out that failure. Going to try downgrading `insta` I guess?

**Comment by chanman3388** [comment]
> Can't for the life of me figure out that failure. Going to try downgrading `insta` I guess?

I think I figured it out, I put the wrong path inside of `src/rules/pylint/mod.rs`

File: src/rules/pylint/rules/too_many_statements.rs [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+                count += 1;
+                count += get_num_statements(body);
+                if !orelse.is_empty() {
+                    count += 1 + get_num_statements(orelse);
+                }
+                if !finalbody.is_empty() {
+                    // weird, but in making samples for pylint, the finally statement counts as 2
+                    count += 2 + get_num_statements(finalbody);
+                }
+                println!("{} {} {}", body.len(), orelse.len(), finalbody.len());
```
Comment (charliermarsh) [inline src/rules/pylint/rules/too_many_statements.rs:79]: Nit: extra print here.

File: src/rules/pylint/rules/too_many_statements.rs [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+
+#[cfg(test)]
+mod tests {
+    use anyhow::Result;
+    use rustpython_parser::parser;
+
+    use super::get_num_statements;
+
+    #[test]
+    fn test_pass() -> Result<()> {
```
Comment (charliermarsh) [inline src/rules/pylint/rules/too_many_statements.rs:133]: Thank you for all the tests.

File: src/rules/pylint/rules/too_many_statements.rs [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+                orelse,
+                finalbody,
+            } => {
+                count += 1;
+                count += get_num_statements(body);
+                if !orelse.is_empty() {
+                    count += 1 + get_num_statements(orelse);
+                }
+                if !finalbody.is_empty() {
+                    // weird, but in making samples for pylint, the finally statement counts as 2
```
Comment (charliermarsh) [inline src/rules/pylint/rules/too_many_statements.rs:76]: Did you review the Pylint source when writing this? Or was it all inferred from the tests?

Comment (chanman3388) [reply src/rules/pylint/rules/too_many_statements.rs:76]: I made a load of examples by hand and ran pylint on them to see what was going on.

Comment (chanman3388) [reply src/rules/pylint/rules/too_many_statements.rs:76]: Hm, the architecture for pylint is slightly different however:
```
def visit_tryfinally(self, node: nodes.TryFinally) -> None:
    """Increments the branches counter."""
    self._inc_branch(node, 2)
    self._inc_all_stmts(2)
```

**Pull request merged**